### PR TITLE
Add the option to create a PlaylistElementVM based on a TimelineEventVM

### PR DESCRIPTION
**Pull request by vmartos**
- This fixes a bug where the eye was not visible due to creation of other TimelineEventVM Instance

#### Things to review:
- [x] Unit tests
- [x] No hardcoded resources (strings, sizes, colors...)
- [x] Sorted usings
- [ ] Add summaries where needed 

If needed, you can add new checkboxes to https://github.com/fluendo/VAS/blob/master/.github/PULL_REQUEST_TEMPLATE.md

### CODE REVIEW

File: VAS.Core/ViewModel/PlaylistElementVM.cs [hunk 1]
```diff
@@ -73,7 +73,7 @@ public class PlaylistElementVM : ViewModelBase<IPlaylistElement>, IPlayable
 		/// Gets or sets a value indicating whether this <see cref="T:VAS.Core.ViewModel.PlaylistElementVM"/> is playing.
 		/// </summary>
 		/// <value><c>true</c> if playing; otherwise, <c>false</c>.</value>
-		public bool Playing { get; set; }
+		public virtual bool Playing { get; set; }
```
Comment (VoldarGames) [inline VAS.Core/ViewModel/PlaylistElementVM.cs]: Why this needs to be virtual, instead of using a simple property?

Comment (VoldarGames) [reply VAS.Core/ViewModel/PlaylistElementVM.cs]: in case it is needed, please summarize it

Comment (vmartos) [reply VAS.Core/ViewModel/PlaylistElementVM.cs]: because PlaylistPlayElementVM needs to override this property to change the TimelineEventVM Playing property. Because the TreeViews are using TimelineEventVM's to render cells, so for PlaylistPlayElementVM we need to use the TimelineEventVM.Playing property, not for the rest of PlaylistElementVM

Comment (VoldarGames) [reply VAS.Core/ViewModel/PlaylistElementVM.cs]: 👍 can you summarize it?

File: VAS.Core/Interfaces/IViewModelFactoryService.cs [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+	/// An interface for creating correctly typed ViewModel instances based on it's model, this is useful to work with
+	/// base classes and create child viewmodels without knowing the reference on it.
+	/// </summary>
+	public interface IViewModelFactoryService
+	{
+		/// <summary>
+		/// Gets the TypeMappings Dictionary, where Key is the Model Type and Value is the ViewModel Type
+		/// </summary>
+		/// <value>The Type Mappings</value>
+		Dictionary<Type, Type> TypeMappings { get; }
```
Comment (ylatuya) [inline VAS.Core/Interfaces/IViewModelFactoryService.cs]: Is this required to be public now?
Returning a dicctionary any one can mess with it and you don't really know how to use it. Instead you can have a public method AddMapping that internally uses that diccionary.
